### ISSUE TITLE
cache-api dependency problem

### ISSUE THREAD

**Issue by trantienduchn** [ical4j/ical4j#250]
**Describe the bug**
using `ical4j` requires `javax.cache:cache-api` dependency in classpath while `javax.cache:cache-api` is declared as an optional dependency in `ical4j` pom file. 

We are using 
```
        <dependency>
            <groupId>org.mnode.ical4j</groupId>
            <artifactId>ical4j</artifactId>
            <version>3.0.0</version>
        </dependency>
```
and in `ical4j` pom we see
```
    <dependency>
      <groupId>javax.cache</groupId>
      <artifactId>cache-api</artifactId>
      <version>1.0.0</version>
      <scope>compile</scope>
      <optional>true</optional>
    </dependency>
```

When ran:
```java
Calendar calendar = new CalendarBuilder().build(anInputStream);
```
Got exception of missing `cache-api` dependency:
```
Exception in thread "main" java.lang.BootstrapMethodError: java.lang.IllegalAccessError: no such constructor: net.fortuna.ical4j.util.JCacheTimeZoneCache.<init>()void/newInvokeSpecial
	at net.fortuna.ical4j.model.TimeZoneLoader.cacheInit(TimeZoneLoader.java:275)
	at net.fortuna.ical4j.model.TimeZoneLoader.<init>(TimeZoneLoader.java:81)
	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.<init>(TimeZoneRegistryImpl.java:125)
	at net.fortuna.ical4j.model.TimeZoneRegistryImpl.<init>(TimeZoneRegistryImpl.java:116)
	at net.fortuna.ical4j.model.DefaultTimeZoneRegistryFactory.createRegistry(DefaultTimeZoneRegistryFactory.java:48)
	at net.fortuna.ical4j.data.CalendarBuilder.<init>(CalendarBuilder.java:105)
Caused by: java.lang.IllegalAccessError: no such constructor: net.fortuna.ical4j.util.JCacheTimeZoneCache.<init>()void/newInvokeSpecial
	at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:483)
	... 7 more
Caused by: java.lang.NoClassDefFoundError: javax/cache/configuration/Configuration
	at java.lang.invoke.MethodHandleNatives.resolve(Native Method)
	at java.lang.invoke.MemberName$Factory.resolve(MemberName.java:975)
	at java.lang.invoke.MemberName$Factory.resolveOrFail(MemberName.java:1000)
	at java.lang.invoke.MethodHandles$Lookup.resolveOrFail(MethodHandles.java:1394)
	at java.lang.invoke.MethodHandles$Lookup.linkMethodHandleConstant(MethodHandles.java:1750)
	at java.lang.invoke.MethodHandleNatives.linkMethodHandleConstant(MethodHandleNatives.java:477)
	... 7 more
Caused by: java.lang.ClassNotFoundException: javax.cache.configuration.Configuration
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 13 more
```

**To Reproduce**
1. Create new maven project
2. add dependency:
```
        <dependency>
            <groupId>org.mnode.ical4j</groupId>
            <artifactId>ical4j</artifactId>
            <version>3.0.0</version>
        </dependency>
```

3. run the statement `Calendar calendar = new CalendarBuilder().build(anInputStream);`

**Expected behavior**
`cache-api` is declared not as an optional dependency in `ical4j` pom file

**Environment (please complete the following information):**
 - OS: linux, mac os
 - Java Version: 8

**Comment by benfortuna** [comment]
A fix has just been applied in #245 - check for latest release shortly.

**Comment by benfortuna** [comment]
Fix applied. Closing issue.
